Fix read_file for shapefiles without image_path/label columns (#997)

SuMayaBee opened the pull request:
## Description

Allow users to pass `image_path` and `label` arguments to `read_file()` when reading shapefiles that don't have these columns.

### Problem

When users have a shapefile without `image_path` or `label` columns, `read_file()` crashes with:
```
ValueError: No image_path column found in shapefile, please specify rgb path
```

### Solution

1. **Pass `image_path` argument to `shapefile_to_annotations()`** - Modified `utilities.py` to forward the `image_path` argument when reading shapefiles.

2. **Add warning when `image_path` is passed** - Users see a warning confirming the value will be assigned to every row.

3. **Allow `label` argument** - Removed blocking error for missing label column. Now defaults to "Unknown" if not provided.

4. **Write tests for shapefiles** - Added 4 tests covering all scenarios.

5. **Add documentation example** - Added example in `docs/user_guide/01_Reading_data.md` with argument table showing required vs optional parameters.

### Before & After

**For shapefiles without `image_path` and `label` columns:**

| # | Scenario | Before | After |
|---|----------|--------|-------|
| 1 | Pass only `image_path` argument | ❌ `ValueError` | ✅ Works, label defaults to "Unknown" |
| 2 | Pass both `image_path` and `label` arguments | ❌ `ValueError` | ✅ Works |

**For shapefiles with `image_path` and `label` columns:**

| # | Scenario | Before | After |
|---|----------|--------|-------|
| 1 | No arguments passed | ✅ Works | ✅ Works (reads from columns) |
| 2 | Pass `image_path` argument | ✅ Works | ✅ Works (uses argument, overrides column) |
| 3 | Pass `label` argument | ✅ Works | ✅ Works (uses argument, overrides column) |
| 4 | Pass both arguments | ✅ Works | ✅ Works (uses arguments, overrides columns) |

### Usage Improvement

**Previously (workaround):**
```python
raster_path = "/path/to/image.tif"
gdf = gpd.read_file("/path/to/annotations.shp")
gdf["image_path"] = os.path.basename(raster_path)
gdf["label"] = "Tree"
ground_truth = read_file(gdf, root_dir=os.path.dirname(raster_path))
```
This required two extra library imports (`os` and `geopandas`).

**Now:**
```python
ground_truth = read_file("/path/to/annotations.shp", image_path="/path/to/image.tif", label="Tree")
```
No extra library imports needed. Here, `root_dir` is optional when `image_path` is a full path. `label` is optional too and defaults to "Unknown" when not provided.

## Related Issue(s)

Closes #997

## AI-Assisted Development

- [x] I used AI tools (e.g., GitHub Copilot, ChatGPT, etc.) in developing this PR
- [x] I understand all the code I'm submitting

SuMayaBee committed:
Fix read_file for shapefiles without image_path/label columns (#997)

SuMayaBee committed:
Refactor shapefile tests to use pytest fixtures

henrykironde reviewed (changes requested):
@SuMayaBee Thanks for your perspective on the issue. The function you’re modifying has changed; could you rebase and update this PR so we can confirm whether these changes are still needed?

SuMayaBee committed:
Chores: Kept the latest changes in the test utitlities

SuMayaBee committed:
Fix label argument not overriding existing label column in read_file

SuMayaBee commented:
@henrykironde Thanks! I've rebased onto upstream/main and updated the implementation to work with the new function structure. The changes are still needed - I've integrated them and fixed a bug where label override wasn't being applied. All tests pass. Ready for review!

SuMayaBee commented:
Hi @henrykironde! Just a quick follow-up on this. I’ve rebased and updated the implementation as discussed. Whenever you have a moment, I’d love to get your thoughts so we can move this forward.

bw4sz commented:
I think we want to solve this issue, but I don't believe this PR is the place, the tests are too much to add for such a small addition, the docs are too long, this feels like alot of AI code. @SuMayaBee you are welcome to open a new PR if you'd like, but i'm going to close this one.
